File: frontend/src/app/model/BrowserResultDTO.ts
import { ExtendedPostDTO } from "./ExtendedPostDTO";
import { ThreadDTO } from "./ThreadDTO";
import { UserProfileDTO } from "./UserProfileDTO";

export interface BrowserResultDTO {
    foundThreads: ThreadDTO[],
    foundPosts: ExtendedPostDTO[],
    foundUserProfiles: UserProfileDTO[]
}

File: backend/dto/src/main/java/pl/dawid0604/pcForum/dto/BrowserResultDTO.java
package pl.dawid0604.pcForum.dto;

import pl.dawid0604.pcForum.dto.post.ExtendedPostDTO;
import pl.dawid0604.pcForum.dto.thread.ThreadDTO;
import pl.dawid0604.pcForum.dto.user.UserProfileDTO;

import java.util.List;

public record BrowserResultDTO(List<ThreadDTO> foundThreads, List<ExtendedPostDTO> foundPosts,
                               List<UserProfileDTO> foundUserProfiles) { }
